mlflow/tracking/fluent.py

```diff
@@ -278,15 +279,15 @@ def start_run(
 
         exp_id_for_run = experiment_id if experiment_id is not None else _get_experiment_id()
 
-        user_specified_tags = tags or {}
+        user_specified_tags = deepcopy(tags) or {}
         if parent_run_id is not None:
             user_specified_tags[MLFLOW_PARENT_RUN_ID] = parent_run_id
         if run_name is not None:
             user_specified_tags[MLFLOW_RUN_NAME] = run_name
 
-        tags = context_registry.resolve_tags(user_specified_tags)
+        run_tags = context_registry.resolve_tags(user_specified_tags)
```

Comment:
```suggestion
        resolved_tags = context_registry.resolve_tags(user_specified_tags)
```

Reply:
Renamed the variable and changed the line just after that!